### PR TITLE
feat: support zeebe:TaskListener

### PR DESCRIPTION
**Pull request by misiekhardcore**
This adds support for `zeebe:TaskListener` contained in `zeebe:TaskListeners` container.

related to https://github.com/camunda/camunda-modeler/issues/4590

### CODE REVIEW

**Comment by barmac** [comment]
Please have a look closely at the XML example from https://github.com/camunda/camunda-modeler/issues/4590#issue-2573661214
We need the element names to exactly match the XML so that the engine can understand it. Note that this project (zeebe-bpmn-moddle) is only used on the frontend while the engine parses BPMN with its own BPMN model.

**Comment by misiekhardcore** [comment]
@barmac I see 👍🏽 I was mislead by the issue name and blindly followed `User task listener`

File: test/fixtures/xml/zeebe-taskListeners.bpmn
Comment (barmac) [inline test/fixtures/xml/zeebe-taskListeners.bpmn:8]: Let's make sure the fixture matches what we want to implement, i.e. only the User Task should contain the task listeners. Otherwise it's confusing as the properties on the process won't be editable.

```suggestion
```

File: test/fixtures/xml/zeebe-taskListeners.bpmn
Comment (barmac) [inline test/fixtures/xml/zeebe-taskListeners.bpmn:21]:
```suggestion
```

File: test/fixtures/xml/zeebe-taskListeners.bpmn
Comment (barmac) [inline test/fixtures/xml/zeebe-taskListeners.bpmn:41]: Is this a copy paste result? The user task looks funny ;)

image

File: test/fixtures/xml/zeebe-taskListeners.bpmn
Comment (barmac) [inline test/fixtures/xml/zeebe-taskListeners.bpmn:22]:
```suggestion
      <bpmn:userTask id="NestedUserTask">
```

File: test/fixtures/xml/zeebe-taskListeners.bpmn
Comment (barmac) [reply test/fixtures/xml/zeebe-taskListeners.bpmn:41]:
```suggestion
      <bpmndi:BPMNShape id="NestedUserTask_di" bpmnElement="NestedUserTask">
        <dc:Bounds x="200" y="592" width="100" height="80" />
      </bpmndi:BPMNShape>
```

Comment (misiekhardcore) [reply test/fixtures/xml/zeebe-taskListeners.bpmn:41]: yes, this is copy paste result. Should I change it or it doesn't matter?

File: test/fixtures/xml/userTask-zeebe-taskListener.part.bpmn
Comment (barmac) [inline test/fixtures/xml/userTask-zeebe-taskListener.part.bpmn:1]: Let's make this match the expectations. We want the task listeners to be allowed in `bpmn:userTask` only.

Comment (barmac) [reply test/fixtures/xml/userTask-zeebe-taskListener.part.bpmn:1]: BTW how come the test is passing if we read a task here and check for user task as the type?

**Review by barmac — COMMENTED** [review]
A few last steps before we merge this.

File: test/fixtures/xml/zeebe-taskListeners.bpmn
Comment (barmac) [reply test/fixtures/xml/zeebe-taskListeners.bpmn:41]: I left the suggestion so that you can apply it from the UI. This one is not blocking.

**Comment by barmac** [comment]
Feel free to merge this PR once you're done.

**Review by barmac — APPROVED** [review]

File: test/fixtures/xml/userTask-zeebe-taskListener.part.bpmn
Comment (misiekhardcore) [reply test/fixtures/xml/userTask-zeebe-taskListener.part.bpmn:1]: it passes for `bpmn:userTask` and `bpmn:task` but does not for `bpmn:endEvent` for example 🤔

**Pull request merged**
